### PR TITLE
RenderWrap.performLayout should take spacing into consideration

### PR DESCRIPTION
**Pull request by jonahwilliams**
RenderWrap.performLayout  should take spacing into consideration when deciding to move to next row/axis.

Background from #11488:

`renderWrap.performLayout()` checks if the size of the child plus the main axis extend exceeds the main axis limit: `runMainAxisExtent + childMainAxisExtent > maxAxisLimit`. This calculation doesn't involve the spacing, however, so the child could overflow from 0 to spacing.

This should be illustrated by the added test case. For example, the four sized boxes are place in an 800 x 600 container, with widths of 200, 200, 200, 171 and a spacing of 10.  If we go through the numbers, the offsets end up as:

```
0.0  + 200 (width) + 10 (spacing)
210 + 200 (width) + 10 (spacing)
420 + 200 (width) + 10 (spacing)
630 + 171 (width)
801
```
Since this exceeds the containers width of 800, we would expect this box to be moved to the next row.  However, as long as the overflow amount is <= `spacing`, then it will still be placed on the same row. The test case shows with the change added the correct offset is indeed on the next row.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/wrap.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:developer';
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/wrap.dart]: hopefully travis will catch this, but in case it doesn't, you probably want to remove this import

Comment (jonahwilliams) [reply packages/flutter/lib/src/rendering/wrap.dart]: Done

File: packages/flutter/test/widgets/wrap_test.dart [hunk 2]
```diff
@@ -828,4 +828,28 @@ void main() {
     expect(tester.renderObject<RenderBox>(find.byType(Baseline)).size,
            within<Size>(from: const Size(100.0, 200.0), distance: 0.001));
   });
+
+  testWidgets('Spacing with slight overflow', (WidgetTester tester) async {
+
```
Comment (Hixie) [inline packages/flutter/test/widgets/wrap_test.dart]: nit: extraneous blank line

**Comment by Hixie** [comment]
nice!
LGTM

File: packages/flutter/test/widgets/wrap_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter/test/widgets/wrap_test.dart]: Done

**Pull request merged**